src/types/mutex.rs

```diff
@@ -0,0 +1,86 @@
+use std::ops::Deref;
+
+/// Wrapper for [`PyMutex`](https://docs.python.org/3/c-api/init.html#c.PyMutex) exposing an RAII interface.
```

Comment:
Here we should probably advise that the advantage of this is that it automatically detaches from python thread states if necessary.